### PR TITLE
Updating the CI Dockerfile to use the patched HIP and MIOpen versions

### PR DESCRIPTION
**Pull request by deven-amd**
The patched versions are
1. HIP from roc-1.8.x + alternate fix corresponding to HIP PR 457
2. MIOpen from the public master (1.4.2) + fix corresponding to MLOpen PR 1061

Both of the above need to be replaced with their official versions once they become available. This update is a temporary one to make progress on the CI Unit test failures.

Updating the HIP version fixes the following 5 unit tests (removed from the whitelist in this PR)
1. `//tensorflow/python/estimator:estimator_test`
2. `//tensorflow/python/keras:activations_test` 
3. `//tensorflow/python/kernel_tests:losses_test`
4. `//tensorflow/python/kernel_tests:metrics_test`
5. `//tensorflow/python:learning_rate_decay_test`

Updating the MIOpen version fixes the following 12 unit tests (removed from the whitelist in this PR)
1. `//tensorflow/python/keras:normalization_test`
2. `//tensorflow/python/keras:training_gpu_test`
3. `//tensorflow/python/kernel_tests:atrous_conv2d_test`
4. `//tensorflow/python/kernel_tests:conv2d_transpose_test`
5. `//tensorflow/python/kernel_tests:lrn_op_test`
6. `//tensorflow/python/kernel_tests:neon_depthwise_conv_op_test`
7. `//tensorflow/python/ops/parallel_for:gradients_test`
8. `//tensorflow/python/profiler:profiler_test`
9. `//tensorflow/python:cost_analyzer_test`
10. `//tensorflow/python:image_ops_test`
11. `//tensorflow/python:layout_optimizer_test`
12. `//tensorflow/python:memory_optimizer_test`

### CODE REVIEW

**Comment by whchung** [comment]
```
[91mdpkg: dependency problems prevent configuration of miopen-hip:
 miopen-hip depends on miopengemm; however:
  Package miopengemm is not installed.

dpkg: error processing package miopen-hip (--install):
 dependency problems - leaving unconfigured
[0m[91mErrors were encountered while processing:
 miopen-hip
```

**Comment by whchung** [comment]
@deven-amd seems like additional details needed to properly build `MIOpen`?

```
/opt/rocm/lib/libMIOpen.so.1: undefined reference to `MD5'
```

**Comment by deven-amd** [comment]
@whchung 

using an empty string when $HOME is not set, works locally but not in the CI environment. Apparently there are paths for which MIOpen needs to translate `~` to the value `$HOME`. The failures we are seeing are because when using an empty string for `$HOME` those paths route to `/` directory, into which the CI environment does not seem to have write permission into.

So we can now either hardcode $HOME to either $PWD (when $HOME is not set) or figure out a way to set $HOME in `bazel`

As per this bazel issue (https://github.com/bazelbuild/bazel/issues/4401) $HOME ought to be set in bazel, but it very clearly is not. I even tried the simple testcase provided in the isue, and it shows the same thing...can we ping Google on this?

In the meantime I will update the MIOpen workaround in my fork to use $PWD when $HOME is not set, and retest this

**Comment by whchung** [comment]
https://github.com/bazelbuild/bazel/pull/4437

It seems there was a PR trying to add this but was closed for unknown reason? When we’ve got bandwidth perhaps it’s time to recreate one PR for bazel?

**Comment by deven-amd** [comment]
jenkins: restart the test

**Comment by deven-amd** [comment]
jenkins: retest this please
